lib/crypto/crypto_scrypt_smix.c

```diff
@@ -138,7 +138,7 @@ blockmix_salsa8(const uint32_t * Bin, uint32_t * Bout, uint32_t * X, size_t r)
  * Return the result of parsing B_{2r-1} as a little-endian integer.
  */
 static uint64_t
-integerify(const void * B, size_t r)
+integerify(const uint32_t * B, size_t r)
 {
 	const uint32_t * X = (const void *)((uintptr_t)(B) + (2 * r - 1) * 64);
```

Comment:
If we're making `B` a `const uint32_t *` then we might as well fix the pointer arithmetic to avoid the now-unnecessary detour into `uintptr_t`.

Reply:
I wondered about that, and I've uploaded it -- but we can only do this in the plain `_smix.c`.  For `_sse2.c`, we have a `__m128i`, so we still need to do the detour into `uintptr_t` (as far as I know).

Having different code in those two functions doesn't look great:
```
/* plain */
const uint32_t * X = B + (2 * r - 1) * 16;

/* sse2 */
const uint32_t * X = (const void *)((uintptr_t)(B) + (2 * r - 1) * 64);
```

But it's your call.

Reply:
We need to drop it in both places.  If the compiler is going to enforce strict aliasing rules, we have to tell it that X is part of B.  I can't remember the intrinsics for extracting a uint32_t from an __m128i though.